simplecasper/app.py

```diff
@@ -126,5 +148,230 @@ def run(ctx, node_id, console):
+                   'correctly (e.g. with "pyethapp --unlock <acct> account list"). If not, look '
+                   'for automatic backup files in the keystore directory (suffix "~" or '
+                   '"~<number>"). Check the log for further information.')
+        raise
+    click.echo('Account update successful')
+
+
+def unlock_accounts(account_ids, account_service, max_attempts=3, password=None):
+    """Unlock a list of accounts, prompting for passwords one by one if not given.
+
+    If a password is specified, it will be used to unlock all accounts. If not, the user is
+    prompted for one password per account.
+
+    If an account can not be identified or unlocked, an error message is logged and the program
+    exits.
+
+    :param accounts: a list of account identifiers accepted by :meth:`AccountsService.find`
+    :param account_service: the account service managing the given accounts
+    :param max_attempts: maximum number of attempts per account before the unlocking process is
+                         aborted (>= 1), or `None` to allow an arbitrary number of tries
+    :param password: optional password which will be used to unlock the accounts
+    """
+    accounts = []
+    for account_id in account_ids:
+        try:
+            account = account_service.find(account_id)
+        except KeyError:
+            log.fatal('could not find account', identifier=account_id)
+            sys.exit(1)
+        accounts.append(account)
+
+    if password is not None:
+        for identifier, account in zip(account_ids, accounts):
+            try:
+                account.unlock(password)
+            except ValueError:
+                log.fatal('Could not unlock account with password from file',
+                          account_id=identifier)
+                sys.exit(1)
+        return
+
+    max_attempts_str = str(max_attempts) if max_attempts else 'oo'
+    attempt_fmt = '(attempt {{attempt}}/{})'.format(max_attempts_str)
+    first_attempt_fmt = 'Password for account {id} ' + attempt_fmt
+    further_attempts_fmt = 'Wrong password. Please try again ' + attempt_fmt
+
+    for identifier, account in zip(account_ids, accounts):
+        attempt = 1
+        pw = click.prompt(first_attempt_fmt.format(id=identifier, attempt=1), hide_input=True,
+                          default='', show_default=False)
+        while True:
+            attempt += 1
+            try:
+                account.unlock(pw)
+            except ValueError:
+                if max_attempts and attempt > max_attempts:
+                    log.fatal('Too many unlock attempts', attempts=attempt, account_id=identifier)
+                    sys.exit(1)
+                else:
+                    pw = click.prompt(further_attempts_fmt.format(attempt=attempt),
+                                      hide_input=True, default='', show_default=False)
+            else:
+                break
+        assert not account.locked
+
```

Comment:
This block of changes comes directly from https://github.com/ethereum/pyethapp/blob/develop/pyethapp/app.py#L458